core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -166,7 +172,9 @@ private static String addRequestTestCaseAndGetPath(HttpMethod method, Item body,
     String asJson = body != null ? MAPPER.writeValueAsString(body) : null;
     HttpRequest mockRequest = request("/" + path)
         .withMethod(method.name().toUpperCase(Locale.ROOT))
-        .withHeader("Authorization", "Bearer " + BEARER_AUTH_TOKEN);
+        .withHeader("Authorization", "Bearer " + BEARER_AUTH_TOKEN)
+        .withHeader(HTTPClientFactory.CLIENT_VERSION_HEADER, icebergBuildFullVersion)
+        .withHeader(HTTPClientFactory.CLIENT_GIT_COMMIT_SHORT_HEADER, icebergBuildGitCommitShort);
```

Comment:
We add the headers that we expect to see as assertions on the mock request. The actual `HTTPClient` is used, generated by the `HTTPClientFactory`, which injects the headers.

If requests are not generated which have these headers, they will not match and the assertion of this mock request will fail.